models/analytics/water_quality_testing_form_responses_fact.sql

```diff
@@ -0,0 +1,39 @@
+{{ config(materialized='table') }}
+
+WITH water_quality_form_responses AS (
+	SELECT
+        obs.id,
+		obs.date_sample_collection,
+        obs.date_testing,
+        obs.ph_count,
+        obs.chloride_count,
+        obs.hardness,
+        obs.total_alkalinity,
+        obs.bacterial_contamination,
+        obs.nitrate_count,
+        obs.iron_count,
+        obs.arsenic_count,
+        obs.fluoride_count,
+        obs.photos
+		-- Replace with actual logic to join and derive activity_id from activity dim table
+		--         activity_dim.id AS activity_id,
+		-- Replace with actual logic to join and derive location_id from location dim table
+		-- 		location_dim.id AS location_id,
+		obs.num_participants,
+		obs.num_women_participants,
+		obs.num_days_water_unavailable,
+		obs.reasons_water_unavailable,
+		obs.photos
+	FROM
+		{{ ref ('observations_intermediate') }} as obs 
+		--     LEFT JOIN activity_dim ON ...
```

Comment:
You can the dims now right? If you update your branch with master it should be available. Doesn't have the SCD logic as yet but it has the required columns

Reply:
Yes, I have added the dim join for activity. ON activity_type / encounter_type. Location once added in the activity table as we discussed today can be used to join downstream since otherwise there's no key to join encounters with location.